solarwinds/user.go

```diff
@@ -0,0 +1,99 @@
+package solarwinds
+
+import (
+	"fmt"
+	"log"
+)
+
+type User = Invitation
+
+type UserService struct {
+	ActiveUserService *ActiveUserService
+	InvitationService *InvitationService
+}
+
+// Create will create a new invitation for the user. It is not possible to add user without going through invitation.
+func (us *UserService) Create(user User) error {
+	return us.InvitationService.Create(user)
+}
+
+// Update will first try to update an active user with the given email. If no such user exist, will see if there
+// is an invitation with the email, if yes, will revoke the invitation and send a new one. Otherwise, error is returned.
+func (us *UserService) Update(update User) error {
+	activeUser, _ := us.ActiveUserService.GetByEmail(update.Email)
+	if activeUser != nil {
+		activeUserUpdate := UpdateActiveUserRequest{
+			UserId:   activeUser.User.Id,
+			Role:     update.Role,
+			Products: update.Products,
+		}
+		return us.ActiveUserService.Update(activeUserUpdate)
+	}
+
+	log.Printf("Will revoke the invitation and send a new one for user: %v", update.Email)
+	invitationService := us.InvitationService
+	invitationList, err := invitationService.List()
+	if err != nil {
+		return err
+	}
+	var invitationFound bool
+	for _, invitation := range invitationList.Organization.Invitations {
+		if invitation.Email == update.Email {
+			invitationFound = true
+			if err = invitationService.Revoke(update.Email); err != nil {
+				return err
+			}
+		}
+	}
+	if !invitationFound {
+		return fmt.Errorf("there is no invitation with email: %v", update.Email)
+	}
+	if err = invitationService.Create(Invitation{
+		Email:    update.Email,
+		Role:     update.Role,
+		Products: update.Products,
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete will only be effective if it is an invitation. There is no way to delete an active user in Pingdom.
+func (us *UserService) Delete(email string) error {
+	activeUser, _ := us.ActiveUserService.GetByEmail(email)
+	if activeUser != nil {
+		log.Printf("It is not possible to delete an active user.")
```

Comment:
Can we deactivate a user? If so, I think it should be done here. If not, this clause should return an error, not just print a message

Reply:
@mLupine I didn't find a deactivate operation. I have created to custom error so that the client code can make decisions accordingly. Thanks for the advice!